### DISCUSSION TITLE
Maximise across multiple displays

### DISCUSSION THREAD

**Discussion by engegu** [rxhanson/Rectangle#984] [Q&A]
I have a 2 display setup with the second screen set as "extended desktop", is it possible for rectangle to maximise my app window across the whole 2 screen desktop?

**Comment by nertzy** [comment]
I don't think so - modern macOS doesn't actually render windows across multiple screens anymore, right?

Try dragging a window to be half-in and half-out and once you let go of the drag, only one screen is left showing the window.

My educated guess is that Rectangle couldn't change that system-level behavior.

**Reply by rxhanson** [reply]
Thanks for responding on a question that apparently slipped by a while back. 

There is a macOS setting for allowing windows to span multiple displays, but I've never prioritized supporting it. If anyone wants to add support for it, I see no reason to turn down a good pull request for it.

**Reply by ainsleyclark** [reply]
This would be a great use case for me. Recently purchased Pro and using the new Samsung 57". Due to the lack of macOS support on external displays I have to use PBP to treat it as one big display to get 120hz.

Would be amazing to get rectangle to span across these two inputs.

Is there a workaround? How would this be implemented?

**Reply by rxhanson** [reply]
There is currently no workaround. I haven't really looked into implementing it, but here are some quick thoughts:

I suspect there's a global default that corresponds to the System Settings preference for this. You could check that default value to determine if Rectangle should treat multiple displays as one. Rectangle has logic for determining the usable size of the display, and you could update that logic to take into account that System Settings default and change the screen size accordingly. See the `ScreenDetection` & `UsableScreens` classes as a starting point.

**Reply by ainsleyclark** [reply]
Allowing for the windows to overlap on one screen is in `Desktop & Dock` -> `Displays have separate spaces.` Would you suggest to latch on to this preference?
Although I don't have much experience in Swift. Would this be able to become part of the roadmap?

**Reply by ainsleyclark** [reply]
I guess we're looking at (Pseudo)

- If `Displays have separate spaces` is false.
- Obtain screen size of all windows.
- Use `ScreenDetection` to update dimensions?

**Reply by b0mbard1er** [reply]
Just wanted to add my support for this question. Mac support for ultrawide monitors is horrible, running 2 inputs on 1 monitor is the best way to make it work. It would be amazing if it was possible for rectangle to treat 2 screens as if they were 1

**Reply by rxhanson** [reply]
@ainsleyclark maybe that wouldn't be the identical path you'd end up with, but it'd be fine to start there. If I can get some time to look into it, I'll give it a go. If not, I'll let you know and I'd be fine merging in a pull request for this.

**Reply by ainsleyclark** [reply]
That would be fantastic 🙏 it would help me so much.

**Reply by b0mbard1er** [reply]
I started looking into this-- while it's very easy to find the value of `Displays have separate spaces` (screensHaveSeparateSpaces is a property of NSScreens) implementing it without a GUI element seems difficult. The user would need to specify which screens to combine. Say I wanted to treat 2 screens as 1, the naive implementation would only work if no other displays were connected or it would lead to very undesirable results. 

If such a feature were to be implemented I imagine it'd need a toggle and a "displays to combine" selector of some kind.

**Reply by engegu** [reply]
Great to see his getting some traction! 😎

I think my usecase, multiple (2) external displays, is likely to be the most common one. A “maximize across external displays” option, available when displays dont have sepetate spaces, would be great. Arrangement would be as set up in the os?

Hell, i’d even love a “maximize with double width feature”, not as elegant but would work fine for those of us with duals of identical size…

**Reply by ainsleyclark** [reply]
Hey, thank you for starting to look into this 👍 
Just wondering if someone had time to flesh it out by any chance?

**Comment by ainsleyclark** [comment]
Anyone landing, I've created a very hacky Applescript to move windows across multiple screens and assigned it to a keyboard shortcut.

```applescript
set windowTitle to ""
tell application "System Events"
	set windowTitle to name of first application process whose frontmost is true
end tell
tell application "System Events"
	set position of first window of application process windowTitle to {4800, 0}
	set size of first window of application process windowTitle to {1600, 1800}
end tell
```
